src/documentation/orogen/plugins.page

```diff
@@ -29,12 +29,28 @@ seamless integration as it is the only one that has an API for code generation.
 Example of a very simple plugin:
 
 ~~~ ruby
-module OrogenSimplePlugin
-    def add_boolean_attribute(name)
-        add_base_member("simple_plugin", name, "bool")
+class BoolPlugin <  OroGen::Spec::TaskModelExtension
+    attr_accessor :varname
+
+    def early_register_for_generation(task)
+        task.add_base_member("simple_plugin", varname, "bool")
+        task.operations["getBool"].
+            base_body("return #{varname};")
+    end
+end
+
+class Orocos::Spec::TaskContext
+    def add_boolean_attribute(varname)
+        extension_name = "BoolPlugin"
+        if !find_extension(extension_name)
```

Comment:
This is correct. But do we need this functionality in the example? This example basically shows how to setup additional members and methods once.